src/main/java/com/sigopt/model/Checkpoint.java

```diff
@@ -0,0 +1,63 @@
+package com.sigopt.model;
+
+import java.util.*;
+
+public class Checkpoint extends StructObject {
+    public Checkpoint() {
+        super();
+    }
+
+    public Checkpoint(String id) {
+        super();
+        this.set("id", id);
+    }
+
+    public String getId() {
+        return (String) this.get("id");
+    }
+
+    public Integer getCreated() {
+        return Utils.asInteger(this.get("created"));
+    }
+
+    public Metadata getMetadata() {
+        return Utils.mergeInto(new Metadata(), this.get("metadata"));
+    }
+
+    public Boolean getShouldStop() {
+        return (Boolean) this.get("should_stop");
+    }
+
+    public Boolean shouldStop() {
+        return (Boolean) this.get("should_stop");
+    }
+
+    public RichMapObject getStoppingReasons() {
+        return Utils.mergeInto(new RichMapObject(), this.get("stoppping_reasons"));
+    }
+
+    public String getTrainingRun() {
+        return (String) this.get("training_run");
+    }
+
+    public List<MetricEvaluation> getValues() {
+        return Utils.mergeIntoList(new ArrayList<MetricEvaluation>(), this.get("values"), MetricEvaluation.class);
+    }
+
+    public static class Builder extends APIObjectBuilder<Checkpoint> {
+        public Builder() {
+            this.obj = new Checkpoint();
+        }
+
+        public Builder values(List<MetricEvaluation> values) {
+            this.obj.set("values", values);
+            return this;
+        }
+
+        public Builder metadata(Map<String, String> metadata) {
+            this.obj.set("metadata", metadata);
+            return this;
+        }
```

Comment:
This is intentional, generally I don't see why we want builders for fields that clients can't set. For example created for observations is in the builder but if you actually use it will error out as its a server-set field.

Reply:
ideally we avoid encoding serverside logic in the client libraries

- to avoid duplicating logic (between server/client, and across multiple clients)
- so that the clients are as thin as possible
- for maximum forwards/backwards compatibility (what fields we allow/disallow changing could change in the future, or could be dependent on more complicated logic)

Reply:
As user I would be pretty frustrated if an API client exposed functions to me that would always cause an error.

The only advantage I see is if we go from something that was once only set serverside to allowing it to being set clientside then we don't have to update the client.

I don't see that as something that would/has happen/d super frequently?

Reply:
To throw some light on this, I think that most users copy paste from examples and the docs, rather than digging into the code of the client themselves. From what I've seen of how they find out how to use the clients, it's unlikely they'd discover some hidden functionality of the client, like what you're describing as a field that would break.